### DISCUSSION TITLE
USB Audio Class - an issue with the TinyUSB implementation

### DISCUSSION THREAD

**Discussion by h5n1xp** [earlephilhower/arduino-pico#2709] [General]
I can seem to get the USB to setup as an Audio Class device... Looking through the AdaFruit TinyUSB source, it's not entirely clear that this supported?

**Comment by earlephilhower** [comment]
I don't think the Adafruit library has audio support, and Tiny USB needs to be reconfigured and libpico rebuilt to enable it.  At that point you could use native Tiny USB cams to manage it.  There's an example in the pico-extras repo if I remember correctly.

**Comment by h5n1xp** [comment]
I have found this fork of TinyUSB which does have audio enabled.

https://github.com/pschatzmann/Adafruit_TinyUSB_Arduino


By adding the line below to my platformio.ini file I was able to get the Audio class working in TinyUSB:

lib_deps =   https://github.com/pschatzmann/Adafruit_TinyUSB_Arduino.git#Audio


At first it caused some missing header file errors (SPI.h?!), but after a few attempts it worked. I hope others find this helpful.

Many thanks for all your hard work @earlephilhower
